backend/wmg/pipeline/summary_cubes/calculate_markers.py

```diff
@@ -145,12 +115,120 @@ def _query_tiledb(
     agg = query.groupby(gb_dims_es).sum()
     n_cells = cell_counts_query.groupby(gb_dims).sum()["n_total_cells"]
 
-    keep_dataset_ids = group_by_dims is None and "dataset_id" in filters.keys()
-    if genes is None:
-        genes = list(agg.index.levels[0])
+    genes = list(agg.index.levels[0])
+    n_cells_per_gene, n_cells_index = _calculate_true_n_cells(n_cells, genes, dataset_to_gene_ids, keep_dataset_ids)
+
+    # roll up across cell type descendants
+    if "cell_type_ontology_term_id" in agg.index.names:
+        # make tidy
+        for name in agg.index.names:
+            agg[name] = agg.index.get_level_values(name)
+        agg = rollup_across_cell_type_descendants(agg)
+        # remove tidy
+        agg.drop(columns=agg.index.names, inplace=True)
+
+    return agg, n_cells_per_gene, n_cells_index, genes
+
+
+def _query_tiledb_context(
```

Comment:
This function wraps `_query_tiledb_context_memoized`. The reason I did not memoize this function directly is because the context query requires both target_filters and context_filters to know what the group-by dimensions should be. The `cell_type_ontology_term_id` in target_filters will change frequently (whereas the group-by dims will often stay the same) which would break memoization if it were an argument to the memoized function. `_query_tiledb_context_memoized` only needs the context_filters so its memoization will not be broken as often.

In the future, if this code were to be used to calculate marker genes for arbitrary filters, this could probably be further refactored to get better memoization, but that is out of scope for now.